File: src/main/java/org/jukeboxmc/block/BlockCaveVines.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemCaveVines;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockCaveVines extends Block{

    public BlockCaveVines() {
        super( "minecraft:cave_vines" );
    }

    @Override
    public ItemCaveVines toItem() {
        return new ItemCaveVines();
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.CAVE_VINES;
    }

    public void setGrowingPlantAge( int value ) {
        this.setState( "growing_plant_age", value );
    }

    public int getGrowingPlantAge() {
        return this.stateExists( "growing_plant_age" ) ? this.getIntState( "growing_plant_age" ) : 0;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockCaveVines.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.CaveVines
import org.jukeboxmc.server.block.JukeboxBlock

class BlockCaveVines(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates),
    CaveVines {

    override fun getGrowingPlantAge(): Int {
        return this.getIntState("growing_plant_age")
    }

    override fun setGrowingPlantAge(value: Int): BlockCaveVines {
        return this.setState("growing_plant_age", value)
    }
}